APIServer/appscale/api_server/app_identity.py

```diff
@@ -130,19 +136,55 @@ def get_access_token(self, scopes, service_account_id=None,
         Raises:
             UnknownError if the service account is not configured.
         """
-        if self._key is None:
```

Comment:
Moving it to https://github.com/AppScale/appscale/blob/cd3bcc68ccaefa66af4e0366cfcdaea49e7664d7/APIServer/appscale/api_server/app_identity.py#L182-L183 seems that `if service_account_name is not None` then `self._key is not None` anyway?

Reply:
Good catch. I modified it a bit so that a user can still get an access token even if `self._key` is None by specifying a custom key name.